app/screens/reaction_list/reaction_list.js

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {View} from 'react-native';
+import {intlShape} from 'react-intl';
+import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view';
+
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+import {
+    generateUserProfilesById,
+    getMissingUserIds,
+    getReactionsByName,
+    getSortedReactionsForHeader,
+    sortReactions,
+} from 'app/utils/reaction';
+
+import ReactionHeader from './reaction_header';
+import ReactionRow from './reaction_row';
+
+import {ALL_EMOJIS} from 'app/constants/emoji';
+
+export default class ReactionList extends PureComponent {
+    static propTypes = {
+        actions: PropTypes.shape({
+            getMissingProfilesByIds: PropTypes.func.isRequired,
+        }).isRequired,
+        allUserIds: PropTypes.array,
+        navigator: PropTypes.object,
+        reactions: PropTypes.array.isRequired,
+        theme: PropTypes.object.isRequired,
+        teammateNameDisplay: PropTypes.string,
+        userProfiles: PropTypes.array,
+    };
+
+    static contextTypes = {
+        intl: intlShape.isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+        const {allUserIds, reactions, userProfiles} = props;
+
+        this.userProfilesById = generateUserProfilesById(userProfiles);
+        this.getMissingProfiles(allUserIds, userProfiles);
+
+        const reactionsByName = getReactionsByName(reactions);
+
+        this.state = {
+            selected: ALL_EMOJIS,
+            reactions,
+            reactionsByName,
+            sortedReactions: sortReactions(reactionsByName),
+            sortedReactionsForHeader: getSortedReactionsForHeader(reactionsByName),
+        };
+
+        props.navigator.setOnNavigatorEvent(this.onNavigatorEvent);
+    }
+
+    static getDerivedStateFromProps(nextProps, prevState) {
+        if (!nextProps.reactions.length !== prevState.reactions.length) {
+            const {reactions} = nextProps;
+
+            const reactionsByName = getReactionsByName(reactions);
+
+            return {
+                reactions,
+                reactionsByName,
+                sortedReactions: sortReactions(reactionsByName),
+                sortedReactionsForHeader: getSortedReactionsForHeader(reactionsByName),
+            };
+        }
+
+        return null;
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.userProfiles.length !== this.props.userProfiles.length) {
```

Comment:
I don't think we need to specifically check length here. Checking `prevProps.userProfiles !== this.props.userProfiles` should be enough here